x/evm/statedb/statedb.go

```diff
@@ -53,9 +51,18 @@ var _ vm.StateDB = &StateDB{}
 // * Contracts
 // * Accounts
 type StateDB struct {
-	keeper   Keeper
-	ctx      sdk.Context
-	cacheCtx sdk.Context
+	keeper Keeper
+	// origCtx is the context passed in by the caller
+	origCtx sdk.Context
+	// ctx is a branched context on top of the context the caller passes in
+	ctx sdk.Context
+	// cacheMS caches the `ctx.MultiStore()` to avoid type assertions all the time.
+	cacheMS cachemulti.Store
+
+	// the action to commit native state, there are two cases:
+	// if the parent store is not `cachemulti.Store`, we create a new one, and call `Write` to commit.
```

Comment:
```suggestion
	// if the parent store is not `cachemulti.Store`, we create a new one, and call `Write` to commit, this could only happen in unit tests.
```